agent-bootstrap/src/main/java/io/opentelemetry/auto/typedspan/HttpServerSemanticConvention.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentelemetry.auto.typedspan;
+
+import io.opentelemetry.trace.Span;
+
+public interface HttpServerSemanticConvention {
```

Comment:
Are you sure this should not extend `HttpSemanticConvention`?

Reply:
The problem is that if a user creates an HttpServerSpan and calls an HttpSpan method, it does not have visibility any more to the setters for HttpServer attributes.

Reply:
Sorry, I don't understand.

Reply:
I will use a small snippet to showcase the problem.

```java
public interface HttpSemanticConvention {
   HttpSemanticConvention setHttpMethod(String httpMethod);
   HttpSemanticConvention setHttpUrl(String httpUrl);
}
public interface HttpServerSemanticConvention extends HttpSemanticConvention {
   HttpServerSemanticConvention setHttpServerName(String httpServerName);
   HttpServerSemanticConvention setHttpRoute(String httpRoute)
}

public class Example {
 void m(){
	HttpServerSemanticConvention span = createHttpServerSpan();
    span
	.setHttpServerName("my server) // Returns an object that sees Http and HttpServer methods
    .setHttpUrl("my URL") // Returns an object that sees only Http methods
    .setHttpMethod("my method") // this is the only method I can call after setHttpUrl
 }

}

```

Reply:
Got it. Let it be so then :)